### DISCUSSION TITLE
Outputting sub app results to different directory

### DISCUSSION THREAD

**Discussion by bhollra** [idaholab/moose#17732] [Q&A General] [answered]
Hi MOOSE community,

I am running a simulation with several similar sub applications. My MultiApps block references the same input file for each subapp, and uses the positions_file parameter to specify the location of each.

It would be helpful for me if the exodus and csv results from these sub apps were output into a separate directory within the main simulation directory, so that after running my simulation the directory will look like:
-main/
----main.i
----main.e
----main.csv
----subapp/
--------subapp.i
--------main_out_subapp0_out.e
--------main_out_subapp0_out.csv
--------main_out_subapp1_out.e
--------main_out_subapp1_out.csv
--------main_out_subapp2_out.e
--------main_out_subapp2_out.csv

I tired using the file_base parameter in the subapp input file, but this resulted in each subapp output overwriting the last.

Is this possible to do within MOOSE, or is moving the results via the command line after the simulation has finished my best option?

Thanks!
Brent

**Comment by YaqiWang** [comment, marked as answer]
So if you set the `Outputs/file_base` in the MultiApp input file, only one output file of a type for multiple subapps will be generated? If this is the case, we possibly need to change the code in `MultiApp.C` to create different file_base based on subapp ids.

**Reply by bhollra** [reply]
Yes, for example with:

  [./csv]
    type = CSV
    file_base = subapp/out
  [../]

in the subapp input, only "out.csv" was created in the subapp folder. Having "out{id}.csv" output instead would be exactly what I'm looking for.

**Reply by YaqiWang** [reply]
Oh. Sorry about this. Yes, this is something needs to be fixed in MOOSE. I will create an issue.

**Reply by YaqiWang** [reply]
The issue is here (https://github.com/idaholab/moose/issues/17738). I did not try reproducing it but looks like the code is indeed problematic.

**Reply by YaqiWang** [reply]
In case that the fix for the issue takes too long, you can try change two lines in `MultiApp.C` to temporarily make this work:
```
diff --git a/framework/src/multiapps/MultiApp.C b/framework/src/multiapps/MultiApp.C
index a14da032f1..67deb81904 100644
--- a/framework/src/multiapps/MultiApp.C
+++ b/framework/src/multiapps/MultiApp.C
@@ -775,6 +775,8 @@ MultiApp::createApp(unsigned int i, Real start_time)
   // output base of the master problem and appending the name of the multiapp plus a number to it
   if (app->getOutputFileBase().empty())
     app->setOutputFileBase(_app.getOutputFileBase() + "_" + multiapp_name.str());
+  else if (_my_num_apps > 1)
+    app->setOutputFileBase(app->getOutputFileBase() + "_" + multiapp_name.str());
   preRunInputFile();
   app->runInputFile();
```
Best.
